Library/Homebrew/cask/dsl.rb

```diff
@@ -352,6 +353,25 @@ def arch(arm: nil, intel: nil)
       end
     end
 
+    # Sets the cask's os strings.
+    #
+    # ### Example
+    #
+    # ```ruby
+    # os macos: "darwin", linux: "tux"
+    # ```
+    #
+    # @api public
+    def os(macos: nil, linux: nil)
```

Comment:
Would be nice to use a `linux: nil` state here to infer "this cask does not support Linux", at least in certain cases, rather than needing thousands of `depends_on :macos` CC @p-linnane for thoughts

Reply:
The problem is that if you use an `on_linux` block this assumption breaks

Reply:
@SMillerDev could be if neither of those are used.

Reply:
I think I'll need @Rylan12 to help me with that one. Will probably be somewhere during FOSDEM.